### ISSUE TITLE
МГТС Московская область: нету рабочих стратегий для Дискорда по http3

### ISSUE THREAD

**Issue by antieksi** [bol-van/zapret#482]
```
* SUMMARY
ipv4 discord.com curl_test_http : working without bypass
ipv4 discord.com curl_test_https_tls12 : winws --wf-l3=ipv4 --wf-tcp=443 --dpi-desync=syndata
ipv4 discord.com curl_test_http3 : winws not working
```

Уже второй раз запускаю скрипт `blockcheck` с таким результатом. Выставляю себе вот такие аргументы запуска с консоли:

```
start "zapret: http,https,quic" /min "%~dp0winws.exe" ^
--wf-l3=ipv4 --wf-tcp=443-65535 --wf-udp=443-65535 ^
--filter-tcp=443 --hostlist="%~dp0list-discord.txt" --dpi-desync=syndata --dpi-desync-autottl=2 --dpi-desync-fooling=md5sig --dpi-desync-fake-tls="%~dp0tls_clienthello_www_google_com.bin" ^
--filter-udp=443 --hostlist="%~dp0list-discord.txt" --dpi-desync=syndata --dpi-desync-udplen-increment=10 --dpi-desync-repeats=6 --dpi-desync-udplen-pattern=0xDEADBEEF --dpi-desync-fake-quic="%~dp0quic_initial_www_google_com.bin" --new ^
--filter-udp=50000-65535 --dpi-desync=syndata --dpi-desync-any-protocol --dpi-desync-fake-quic="%~dp0quic_initial_www_google_com.bin" --new
```

Конечно, единожды получается зайти на заглавную страницу Дискорда, но не в сами чаты (подозреваю, что как раз из-за `winws not working`), после чего и заглавная страница отваливается. Это проблема конфигурации или провайдер уж слишком умный?

**Comment by inkcmd** [comment]
start "zapret: discord" /min "%BIN%winws.exe" ^
--wf-tcp=80,443 --wf-udp=443,50000-65535 ^
--filter-udp=443 --hostlist="%~dp0list-general.txt" --dpi-desync=fake --dpi-desync-udplen-increment=10 --dpi-desync-repeats=6 --dpi-desync-udplen-pattern=0xDEADBEEF --dpi-desync-fake-quic="%BIN%quic_initial_www_google_com.bin" --new ^
--filter-udp=50000-65535 --dpi-desync=fake,tamper --dpi-desync-any-protocol --dpi-desync-fake-quic="%BIN%quic_initial_www_google_com.bin" --new ^
--filter-tcp=80 --hostlist="%~dp0list-general.txt" --dpi-desync=fake,split2 --dpi-desync-autottl=2 --dpi-desync-fooling=md5sig --new ^
--filter-tcp=443 --hostlist="%~dp0list-general.txt" --dpi-desync=fake,split2 --dpi-desync-autottl=2 --dpi-desync-fooling=md5sig --dpi-desync-fake-tls="%BIN%tls_clienthello_www_google_com.bin"

у меня на компе вот такое идеально работает.

**Comment by curdew** [comment]
а как открыть его , просто он моментально закрывается

**Comment by Bestig** [comment]
@inkcmd  автор уже писал что тут много мусора в канале будет
а при такой стретгии гугл сервисы работают у вас, например https://docs.google.com/ ?

**Comment by bol-van** [comment]
quic может и не пробиваться на отдельных провайдерах.
про МГТС много всего писали. там тяжело.

--filter-udp=50000-65535 --dpi-desync=syndata --dpi-desync-any-protocol --dpi-desync-fake-quic="%~dp0quic_initial_www_google_com.bin" --new

вот это полностью бесполезно, поскольку syndata неприменимо к udp. не работает вообще

**Comment by inkcmd** [comment]
> @inkcmd автор уже писал что тут много мусора в канале будет а при такой стретгии гугл сервисы работают у вас, например https://docs.google.com/ ?

да, у меня открывается тот сайт, что вы дали. и я вижу там документы, которые когда то использовал. это под виндой я такое запускал, чтобы хоть что-то работало. 2 месяца искал причину. оказалось, что видосы в браузере грузятся по ipv6 когда он включен в роутере и в адаптере на винде, а я в запрете на роутере поддержку ipv6 не включал. почему то. потом включил и пошло. проверил с выключенным ipv6 везде - тоже работает. чет сразу я и не понял. два месяца заняло

**Comment by Bestig** [comment]
```
set ARGS=--wf-tcp=443 --wf-udp=443 ^
--filter-udp=443 --hostlist=\"%~dp0list-discord.txt\" --dpi-desync=fake --dpi-desync-udplen-increment=10 --dpi-desync-repeats=6 --dpi-desync-udplen-pattern=0xDEADBEEF --dpi-desync-fake-quic=\"%BIN%quic_initial_www_google_com.bin\" --new ^
--filter-tcp=443 --hostlist=\"%~dp0list-discord.txt\" --dpi-desync=fake,split2 --dpi-desync-autottl=2 --dpi-desync-fooling=md5sig --dpi-desync-fake-tls=\"%BIN%tls_clienthello_www_google_com.bin\"
```
вот так точно сработает, ну и с учётом коммента автора, возможно вообще обойтись одним протоколом\портом, а именно:
set ARGS=--wf-tcp=443 ^
--filter-tcp=443 --hostlist=\"%~dp0list-discord.txt\" --dpi-desync=fake,split2 --dpi-desync-autottl=2 --dpi-desync-fooling=md5sig --dpi-desync-fake-tls=\"%BIN%tls_clienthello_www_google_com.bin\"
```

**Comment by antieksi** [comment]
Опробовал стратегии от @Bestig и @inkcmd, ни одна так и не сработала.

**Comment by Elllkere** [comment]
```
--wf-tcp=443 --wf-udp=443,50000-50099 --filter-udp=443 --hostlist="%~dp0list-discord.txt" --dpi-desync=fake --dpi-desync-udplen-increment=10 --dpi-desync-repeats=6 --dpi-desync-udplen-pattern=0xDEADBEEF --dpi-desync-fake-quic="%~dp0quic_initial_www_google_com.bin" --new ^
--filter-udp=50000-50099 --ipset="%~dp0list-discord-ip.txt" --dpi-desync=fake --dpi-desync-any-protocol --dpi-desync-fake-quic="%~dp0quic_initial_www_google_com.bin" --new ^
--filter-tcp=443 --hostlist="%~dp0list-discord.txt\" --dpi-desync=fake,split2 --dpi-desync-autottl=2 --dpi-desync-fooling=md5sig --dpi-desync-fake-tls="%~dp0tls_clienthello_www_google_com.bin
```

**Comment by bol-van** [comment]
> ipv4 discord.com curl_test_http : working without bypass
> ipv4 discord.com curl_test_https_tls12 : winws --wf-l3=ipv4 --wf-tcp=443 --dpi-desync=syndata
> ipv4 discord.com curl_test_http3 : winws not working

Такое еще может быть, если включен flow offloading на уровне openwrt. В LUCI, например.
